### PR TITLE
add task status named ReleasingFailed

### PR DESCRIPTION
**Pull request by ycfnana**
if pod being terminating for long time because of zombie process，the task will be scheduled this node，and then this job may hang until the pod with zombie process killed force

### CODE REVIEW

**Comment by hwdef** [comment]
How does this PR avoid the problems you raised?

**Comment by ycfnana** [comment]
> How does this PR fix the problems you raised?

    job may be status pipelined when this case happens. if job request resource is future idle, it will be pipelined status, the releasing resource will be used in the way of the future idle.
    I think if the pod has something wrong with termimating process, it should be detected.  the process of pod terminating, will send signal TERM and wait pod.Spec.TerminationGracePeriodSeconds(default 30s) to send signal kill. I think pod terminating has something wrong if wait time exceed TerminationGracePeriodSeconds.
    however, I think this code will result in task status from Releasing to Running, it's not good but I have no idea to chang it to another status

**Comment by hwdef** [comment]
From the code point of view, you just make the conditions for returning releasing more stringent

**Comment by ycfnana** [comment]
> From the code point of view, you just make the conditions for returning releasing more stringent

my fault, its title is  vague. I'll update it
 I met a issue, pod always be Pending when one node has zombie pod and another node has enough reource, due to the relaesing status, I think waiting time larger than  TerminationGracePeriodSeconds it should not be releaseing, may happen something error,  its should be another status.

File: pkg/scheduler/api/helpers.go [hunk 1]
```diff
@@ -33,15 +34,22 @@ func PodKey(pod *v1.Pod) TaskID {
...
+		if pod.DeletionTimestamp != nil &&
+			time.Now().Unix()-pod.DeletionTimestamp.Unix() <= gracePeriodSeconds {
 			return Releasing
 		}
 
 		return Running
 	case v1.PodPending:
-		if pod.DeletionTimestamp != nil {
+		if pod.DeletionTimestamp != nil &&
+			time.Now().Unix()-pod.DeletionTimestamp.Unix() <= gracePeriodSeconds {
```
Comment (hwdef) [inline pkg/scheduler/api/helpers.go:52]: `time.Now().Unix()-pod.DeletionTimestamp.Unix()`will keep growing, If once the time point of less than or equal to is missed, then this judgment will always be false and will never return release. Is this what we want?

Comment (ycfnana) [reply pkg/scheduler/api/helpers.go:52]: it means this pod is always exists with terminating status if `time.Now().Unix()-pod.DeletionTimestamp.Unix()`keep growing, resource cannot be freed. this resource is always allocated for scheduler.

Comment (ycfnana) [reply pkg/scheduler/api/helpers.go:52]: > `time.Now().Unix()-pod.DeletionTimestamp.Unix()`will keep growing, If once the time point of less than or equal to is missed, then this judgment will always be false and will never return release. Is this what we want?

I think gracePeriodSeconds can set bigger,  it should be another status not "Running" status when waiting time bigger than gracePeriodSeconds, but I have no idea what status to set

Comment (hwdef) [reply pkg/scheduler/api/helpers.go:52]: Sorry I don't think this is a perfect fix

Comment (ycfnana) [reply pkg/scheduler/api/helpers.go:52]: I think should add a status  like "ReleaseFailed", in the code of resource calculation, it should be the same as Running. what do you think?

Comment (hwdef) [reply pkg/scheduler/api/helpers.go:52]: @wangyang0616 
Can you help review this PR

Comment (ycfnana) [reply pkg/scheduler/api/helpers.go:52]: yeah, I'll close it and fix it with add staus. when I finish it I'll reopen it again

**Pull request closed without merging**